src/lib/Form/Type/Content/CustomUrl/CustomUrlAddType.php

```diff
@@ -0,0 +1,162 @@
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+class CustomUrlAddType extends AbstractType
+{
+    /** @var LanguageService */
+    private $languageService;
+
+    /** @var URLAliasService */
+    private $urlAliasService;
+
+    /** @var LocationService */
+    private $locationService;
+
+    /** @var ContentService */
+    private $contentService;
+
+    public function __construct(
+        LanguageService $languageService,
+        URLAliasService $urlAliasService,
+        LocationService $locationService,
+        ContentService $contentService
+    ) {
+        $this->languageService = $languageService;
+        $this->urlAliasService = $urlAliasService;
+        $this->locationService = $locationService;
+        $this->contentService = $contentService;
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add(
+                'location',
+                LocationType::class,
+                ['label' => false]
+            )
+            ->add(
+                'path',
+                TextType::class,
+                ['label' => false]
+            )
+            ->add(
+                'language',
+                ChoiceType::class,
+                [
+                    'multiple' => false,
+                    'choice_loader' => new CallbackChoiceLoader(function () {
+                        return $this->languageService->loadLanguages();
+                    }),
+                    'choice_value' => 'languageCode',
+                    'choice_label' => 'name',
+                ]
+            )
+            ->add(
+                'redirect',
+                CheckboxType::class,
+                [
+                    'required' => false,
+                    'label' => false,
+                ]
+            )
+            ->add(
+                'site_root',
+                CheckboxType::class,
+                [
+                    'required' => false,
+                    'label' => false,
+                ]
+            )
+            ->add(
+                'add',
+                SubmitType::class,
+                [
+                    'label' => /** @Desc("Create") */ 'customurl_alias_add_form.add',
+                ]
+            );
+
+        $builder->addEventListener(FormEvents::PRE_SET_DATA, [$this, 'addLanguageSelectorBasedOnContent']);
+        $builder->addEventListener(FormEvents::PRE_SUBMIT, [$this, 'buildFullPathIfRoot']);
+    }
+
+    /**
+     * @param FormEvent $event
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    public function addLanguageSelectorBasedOnContent(FormEvent $event)
+    {
+        /** @var CustomUrlAddData $data */
+        $data = $event->getData();
+        $location = $data->getLocation();
+        if ($location === null) {
```

Comment:
Not a strong requirement but we tend to use yoda comparison `null === $location`